apputils/halo.py

```diff
@@ -16,52 +16,67 @@ def __init__(self):
                                                    self.halo_api_secret_rw)
 
     def list_all_servers_formatted(self):
+        """Return a list of all servers, formatted for Slack."""
         servers = cloudpassage.Server(self.session)
         return fmt.format_list(servers.list_all(), "server_facts")
 
     def list_all_groups_formatted(self):
+        """Return a list of all groups, formatted for Slack."""
         groups = cloudpassage.ServerGroup(self.session)
         return fmt.format_list(groups.list_all(), "group_facts")
 
     def generate_server_report_formatted(self, target):
-        server_id = self.get_id_for_server_target(target)
+        """Return a formatted server report for arg:target server."""
+        server_ids = self.get_id_for_server_target(target)
         result = ""
-        if server_id is not None:
-            print("ServerReport: Starting report for %s" % server_id)
-            server_obj = cloudpassage.Server(self.session)
-            print("ServerReport: Getting server facts")
-            facts = self.flatten_ec2(server_obj.describe(server_id))
-            if "aws_ec2" in facts:
-                result = fmt.format_item(facts, "server_ec2")
-            else:
-                result = fmt.format_item(facts, "server_facts")
-            print("ServerReport: Getting server issues")
-            result += fmt.format_list(self.get_issues_by_server(server_id),
-                                      "issue")
-            print("ServerReport: Getting server events")
-            result += fmt.format_list(self.get_events_by_server(server_id),
-                                      "event")
+        if len(server_ids) == 0:
```

Comment:
Prefer guard clause over if-else conditional.

Example:

server_ids = self.get_id_for_server_target(target)
if not server_ids:
&nbsp;&nbsp;&nbsp;&nbsp;return "Unable to find server %s" % target

perform below loop without the else.

Reply:
OK, NP.